components/browser/state/src/main/java/mozilla/components/browser/state/reducer/MediaSessionReducer.kt

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.reducer
+
+import mozilla.components.browser.state.action.MediaSessionAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.browser.state.state.MediaSessionState
+import mozilla.components.concept.engine.mediasession.MediaSession
+
+internal object MediaSessionReducer {
+    /**
+     * [MediaSessionAction] Reducer function for modifying the [MediaSessionState] of a [SessionState].
+     */
+    fun reduce(state: BrowserState, action: MediaSessionAction): BrowserState {
+        return when (action) {
+            is MediaSessionAction.ActivatedMediaSessionAction ->
+                state.updateMediaSession(action.tabId, action.mediaSession)
+            is MediaSessionAction.DeactivatedMediaSessionAction ->
+                state.updateMediaSession(action.tabId, action.mediaSession)
```

Comment:
Instead of marking the media session as active `true/false`. Could the MediaState be nullable and we set it or remove it? I guess there's no need to keep an `MediaState` around if there's no media session.

Reply:
Sure, I'll make `MediaSessionState` nullable.  When activated, I'll create a new `MediaSessionState`.  When deactivated, I'll set it to null.